src/main/java/com/nike/cerberus/auth/connector/okta/OktaMFAAuthConnector.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2017 Nike, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.nike.cerberus.auth.connector.okta;
+
+import com.google.common.base.Preconditions;
+import com.nike.cerberus.auth.connector.AuthConnector;
+import com.nike.cerberus.auth.connector.AuthData;
+import com.nike.cerberus.auth.connector.AuthMfaDevice;
+import com.nike.cerberus.auth.connector.AuthResponse;
+import com.nike.cerberus.auth.connector.AuthStatus;
+import com.okta.sdk.models.auth.AuthResult;
+import com.okta.sdk.models.factors.Factor;
+import com.okta.sdk.models.usergroups.UserGroup;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.inject.Inject;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Okta Version 1 API implementation of the AuthConnector interface, requires MFA login for all users.
+ */
+public class OktaMFAAuthConnector implements AuthConnector {
+
+    private final OktaAuthHelper oktaAuthHelper;
+
+    @Inject
+    public OktaMFAAuthConnector(final OktaAuthHelper oktaAuthHelper) {
+
+        this.oktaAuthHelper = oktaAuthHelper;
+    }
+
+    @Override
+    public AuthResponse authenticate(String username, String password) {
+
+        final AuthResult authResult = oktaAuthHelper.authenticateUser(username, password, null);
+        final String userId = this.oktaAuthHelper.getUserIdFromAuthResult(authResult);
+        final String userLogin = this.oktaAuthHelper.getUserLoginFromAuthResult(authResult);
+
+        final AuthData authData = new AuthData().setUserId(userId).setUsername(userLogin);
+        final AuthResponse authResponse = new AuthResponse().setData(authData).setStatus(AuthStatus.MFA_REQUIRED);
+
+        final List<Factor> factors;
+
+        if (StringUtils.equals(authResult.getStatus(), OktaAuthHelper.AUTHENTICATION_MFA_REQUIRED_STATUS)
```

Comment:
Nitpick: Rather than StringUtils.equals( ..., ...) this would read nicer if it was encapsulated in a class.

E.g. oktaAuthStatus.isMfaRequired(authResult.getStatus()) or similar